CONTRIBUTING.md

```diff
@@ -0,0 +1,149 @@
+
+### Sprint Reviews and Planning
+
+Approximately every 2 weeks we have sprint reviews and then also a sprint
+planning meeting.
+
+
+
+## How to work on a new feauture/bug
+
+Create an issue on Github, and talk to the team members on the NBIS local-ega
+Slack channel. You can alternatively pick one already created.
+
+Assign yourself to that issue.
+
+Discussions on how to proceed about that issue take place in the comment
+section on that issue, beforehand.
+
+The keyword here is *beforehand*. It is usually a good idea to talk about it
+first. Somebody might have already some pieces in place, we avoid unnecessary
+work duplication and a waste of time and effort.
+
+
+## How we work with Git
+
+All work take place in feature branches. Give your branch a short descriptive
+name and prefix the name with:
+
+   * `feature/` if it is a code feature
+   * `hotfix/` if you are fixing an urgent bug
+
+Use comments in your code, choose variable and function names that clearly show
+what you intend to implement.
+
+Once the feature is done it is merged back into `master` by making a Pull
+Request.
+
+
+### General stuff about git and commit messages
+
+In general it is better to commit often. Small commits are easier to roll back
+and also makes the code easier to review.
+
+Write helpful commit messages that describes the changes and possibly why they
+were necessary.
+
+Each commit should contain changes that are functionally connected and/or
+related. If you for example want to write _and_ in a commit message this is a
+good sign that it should have been two commits.
+
+Learn how to select chunkgs of changed files to do multiple separate commits of
+unrelated things. This can be done with either `git add -p` or `git commit -p`.
+
+
+#### Helpfull commit messages
+
+The commit messages may be seen as meta-comments on the code that are
+incredibly helpful for anyone who wants to know how this piece of software is
+working, including colleagues (current and future) and external users.
+
+Some tips about writing helpful commit messages:
+
+#. Separate subject (the first line of the message) from body with a blank line.
+#. Limit the subject line to 50 characters.
+#. Capitalize the subject line.
+#. Do not end the subject line with a period.
+#. Use the imperative mood in the subject line.
+#. Wrap the body at 72 characters.
+#. Use the body to explain what and why vs. how.
+
+For an in-depth explanation of the above points, please see [How to Write a Git
+Commit Message](http://chris.beams.io/posts/git-commit/).
+
+
+### How we do code reviews
+
+A code review is initiated by making a Pull Request in the appropriate repo on
+github.
+
+Work should not continue on the branch _unless_ it has been marked with the `DO
+NOT MERGE` tag. Once the PR is ready and the `DO NOT MERGE` tag is removed the
+review process can go on to the next stage.
+
+Before review starts it is a good idea to rebase your branch to `master` to
```

Comment:
Is it "a good idea" or is it actually required? I'm personally for a "good idea" approach :)

Reply:
Well, if you rebase before, the test-suite will test what the merge will look like instead of what a merge would have looked like at the graft point. If the branch is rebased before merge conflicts can be resolved before the PR is done as well. For those reasons I think required would be better.

Reply:
Just to add to my previous comment. Both of these activities I mention is work that has to be done, and it's the original developer that knows best how to resolve conflicts. So not rebasing is basically just a way of moving work from now into the future. And if there's a lot of merge conflicts, at least parts of the review will have to be redone resulting in double-work.